### PR TITLE
[SELS-TASK] Implemented Quiz Answering Function

### PR DESCRIPTION
**Pull request by christanplaza**
[Description]: This PR is for the implementation of the Quiz Function. This PR includes the Pagination of the quiz and also restrictions for the user to return to the quiz after they have taken it.

[Commands]
- php artisan migrate:fresh --seed

[Pre-Condition]
- Login as Non-Admin (johndoe@gmail.com, securepassword)
- Go to /categories
- Click on Start lesson
- Answer the quiz. (The first items are the correct answer in this case)

[Expected output]
- There should be a paginated quiz that allows the user to select an answer
- There should be 3 words for the quiz
- After the third word has been answered, the page should redirect to /lessons

[Note]
- There is no results page yet, I am going to be implementing it on the next PR

### CODE REVIEW

File: routes/web.php [hunk 1]
```diff
@@ -32,3 +32,8 @@
 Route::resource('lessons', 'LessonController');
 
 Route::resource('quiz', 'QuizController');
+
+Route::resource('answer', 'AnswerController');
+
+Route::post('/completed-quiz/{quiz}', 'CompletedQuizController@store');
```
Comment (jeremiahC) [inline routes/web.php]: use `group` function for routes to have more organize structure

File: app/Http/Controllers/AnswerController.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            $quiz->result = $quiz->getResult($attributes['lesson_id']);
+            $quiz->save();
+        }
+
+        return redirect($attributes['next_url']);
+    }
+
+    protected function validateAnswer()
+    {
+        return request()->validate([
```
Comment (jeremiahC) [inline app/Http/Controllers/AnswerController.php:30]: You can use Form Request File for validation and you don't need to create a validate function

Comment (christanplaza) [reply app/Http/Controllers/AnswerController.php:30]: This is fixed in c6523a90d5dce97ef45dcb15b970d517f33f88da

File: routes/web.php (hunk 1 above)
Comment (christanplaza) [reply routes/web.php]: I realize that this route is unused in the future. I have now removed it.

**Review by jeremiahC — APPROVED** [review]

**Pull request merged**